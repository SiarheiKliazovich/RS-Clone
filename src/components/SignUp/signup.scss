@import "../../sass/variables";
@import "../../sass/mixins";

.sign__wrapper > {
    box-sizing: border-box;
}

.sign__wrapper {
    display: flex;
    width: 1600px;
    margin: 0 auto;
    justify-content: center;
    align-content: center;
    height: calc(100vh - 200px);
    margin: 0 auto; 
    padding: 0 20px;   

    @include media-desktop {
        width: 1200px;
    }

    @include media-tablet {
        width: 1000px;
    }

    @include media-small-tablet {
        width: 770px;
    }

    @include media-mobile {
        flex-direction: column;
        width: 570px;
    }

    @include media-small-mobile {
        width: 400px;
    }
}

.sign__image {    
    width: calc(50% - 20px);
    display: flex;
    justify-content: center;
    align-content: center;
    margin-right: 20px;

    @include media-mobile {
        width: 100%;
        margin-right: 0;
        margin-bottom: 20px;
        padding-top: 180px;
    }

}

.image-wrapper {
    align-self: center;
    background-image: url("../../assets/images/png/login.png");
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
    width: 100%;
    height: 340px;
}

.sign__content-wrapper {
    display: flex;
    width: calc(50% - 20px);
    justify-content: center;
    align-items: center;
    margin-left: 20px;

    @include media-mobile {
        width: 100%;
        margin-left: 0;
        padding-bottom: 30px;
    }
}

.sign__content {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    width: 100%;
}

.sign__title {
    font-size: $font-size-xxxlg;
    margin-bottom: 15px;

    @include media-small-tablet {
        font-size: $font-size-xxlg;
    }
}

.sign__text {
    font-size: $font-size-lg;
    color: $font-color-dark;
}

.form {
    display: flex;
    flex-direction: column;
}

.field {
    margin-top: 15px;
    outline: none;
    height: 35px;
    border: 1px solid $font-color-light-grey;
    border-radius: 5px;
    padding-left: 5px;
}

.error {
    color: $color-special-1;
    margin-top: 0;
}

.sign__radio {
    margin-top: 15px;

    label {
        cursor: pointer;

        &:first-child {
            margin-right: 30px;
        }        
    }
     
    .radio-field {
        margin-right: 10px; 
        cursor: pointer;
    }
}

.sign__button {
    margin-top: 30px;
    margin-bottom: 15px;
    width: 90px;
    height: 40px;
    color: $font-color-white;
    border: none;
    border-radius: 5px;
    background-color: $color-accent;
    cursor: pointer;

    &:hover,
    &:active {
        background-color: $color-accent-hover;
    }
}

.text-link {
    color: $color-accent;

    &:hover,
    &:active {
        color: $color-accent-hover;
    }
}

.error-tooltip {
    margin-bottom: 30px;
    color: $color-special-1;
}

.message-tooltip {
    margin-bottom: 30px;
    color: $color-accent-green;
}
