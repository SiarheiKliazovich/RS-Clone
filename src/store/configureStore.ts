/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
