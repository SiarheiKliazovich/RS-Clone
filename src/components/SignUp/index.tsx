import { FunctionComponent, useState, useEffect, useRef } from "react";
import { Link, useNavigate } from "react-router-dom";
import { Formik, Field, Form, ErrorMessage } from "formik";
import * as Yup from "yup";
import { useTranslation } from "react-i18next";
import { useSelector } from "react-redux";
import classNames from "classnames";

import { IUser } from "../../interfaces/user";
import { registration } from "../../services/apiAuth";
import Header from "../Header";
import Profile from "../Profile";
import navMenu from "../../data/navmenu";
import { ILangReducer } from "../../interfaces/langReducer";
import useOnClickOutside from "../../hook/useOnClickOutside";
import { IStore } from "../../interfaces/store";
import { nightTheme } from "../../data/constants";

import "./signup.scss";

const SignUp: FunctionComponent = () => {
  const { lang } = useSelector((state: ILangReducer) => state.langReducer);

  const interfaceSettings = useSelector((state: IStore) => state.appInterface);
  const [open, setOpen] = useState(false);
  const {
    accentColor,
    isSidebarFixed,
    isSidebarAccentMode,
    isNavbarNightMode,
    isProfileShow,
    isNightMode,
  } = interfaceSettings;
  const backgroundColor = nightTheme.background.element;

  const ref = useRef<HTMLDivElement>(null);
  useOnClickOutside(ref, () => setOpen(!open), open);

  const list = navMenu.find((c) => c.lang === lang)!;

  const currentURL = window.location.pathname;

  const [message, setMessage] = useState<string>("");
  const [error, setError] = useState<string>("");

  const navigate = useNavigate();

  useEffect(() => {
    setTimeout(() => setMessage(""), 5000);
    setTimeout(() => setError(""), 5000);
  }, [message, error]);

  const { t } = useTranslation("dataLang");

  const click = async (body: Pick<IUser, "email" | "password" | "role">) => {
    try {
      const message = await registration(body);
      setMessage(message);
      setTimeout(() => navigate("/sign-in"), 3000);
    } catch (e) {
      setError(e as string);
    }
  };

  return (
    <>
      <Header
        accentColor={accentColor}
        isNavbarNightMode={isNavbarNightMode}
        isButtonVisible={false}
      >
        <div className="nav__hamburger_wrapper" ref={ref}>
          <span className="nav__hamburger" onClick={() => setOpen(!open)} />
        </div>
        <div
          className={classNames("nav-sidebar", { sidebaropen: open })}
          style={{
            backgroundColor: isNavbarNightMode
              ? backgroundColor
              : accentColor.static,
          }}
        >
          <div className="nav-sidebar-wrapper">
            <ul className="list-nav-sidebar">
              {list.data.map((item) => (
                <li className="nav-sidebar-item" key={item.name}>
                  {currentURL === item.path ? (
                    <span className="active-link">{item.name}</span>
                  ) : (
                    <Link className="nav-sidebar-link" to={item.path}>
                      {item.name}
                    </Link>
                  )}
                </li>
              ))}
            </ul>
          </div>
        </div>
      </Header>
      <Profile
        accentColor={accentColor}
        isProfileShow={isProfileShow}
        isSidebarFixed={isSidebarFixed}
        isSidebarAccentMode={isSidebarAccentMode}
        isNavbarNightMode={isNavbarNightMode}
        isNightMode={isNightMode}
      />
      <div className="sign__wrapper">
        <div className="sign__image">
          <div className="image-wrapper"></div>
        </div>
        <div className="sign__content-wrapper">
          <div className="sign__content">
            <h2 className="sign__title">{t("sign.signUp")}</h2>
            <div className="sign__text">{t("sign.text-2")}</div>
            <Formik
              initialValues={{
                email: "",
                password: "",
                role: "USER",
              }}
              validationSchema={Yup.object({
                email: Yup.string()
                  .email("Incorrect email")
                  .required("Required field"),
                password: Yup.string()
                  .min(4, "Minimum 4 symbols")
                  .max(10, "Maximum 10 symbols")
                  .required("Required field"),
              })}
              onSubmit={(values) => {
                click(values);
              }}
            >
              <Form className="form">
                <Field
                  className="field"
                  id="email"
                  name="email"
                  type="email"
                  placeholder="Email"
                />
                <ErrorMessage className="error" name="email" component="div" />
                <Field
                  className="field"
                  id="password"
                  name="password"
                  type="password"
                  placeholder="Password"
                />
                <ErrorMessage
                  className="error"
                  name="password"
                  component="div"
                />
                <div className="sign__radio">
                  <label>
                    <Field
                      className="radio-field"
                      type="radio"
                      name="role"
                      value="USER"
                    />
                    {t("sign.buyer")}
                  </label>
                  <label>
                    <Field
                      className="radio-field"
                      type="radio"
                      name="role"
                      value="SELLER"
                    />
                    {t("sign.seller")}
                  </label>
                </div>
                <button
                  style={{
                    backgroundColor: isNavbarNightMode
                      ? backgroundColor
                      : accentColor.static,
                  }}
                  className="sign__button"
                  type="submit"
                >
                  {t("sign.signIn")}
                </button>
                {message && <div className="message-tooltip">{message}</div>}
                {error && <div className="error-tooltip">{error}</div>}
              </Form>
            </Formik>
            <div className="sign__text-link">
              {t("sign.have")}{" "}
              <Link
                style={{
                  color: isNavbarNightMode
                    ? backgroundColor
                    : accentColor.static,
                }}
                className="text-link"
                to="/sign-in"
              >
                {t("sign.signIn")}
              </Link>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default SignUp;
