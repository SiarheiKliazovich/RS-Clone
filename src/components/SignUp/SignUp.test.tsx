import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import SignUp from ".";

const mockStore = configureStore([]);

describe("SignUp", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <SignUp />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
