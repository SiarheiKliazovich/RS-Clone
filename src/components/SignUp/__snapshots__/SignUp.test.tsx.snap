// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<div
  class="sign__wrapper"
>
  <div
    class="sign__image"
  >
    <div
      class="image-wrapper"
    />
  </div>
  <div
    class="sign__content-wrapper"
  >
    <div
      class="sign__content"
    >
      <h2
        class="sign__title"
      >
        sign.signUp
      </h2>
      <div
        class="sign__text"
      >
        sign.text-2
      </div>
      <form
        action="#"
        class="form"
      >
        <input
          class="field"
          id="email"
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          class="field"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
        <div
          class="sign__radio"
        >
          <label>
            <input
              checked=""
              class="radio-field"
              name="role"
              type="radio"
              value="USER"
            />
            sign.buyer
          </label>
          <label>
            <input
              class="radio-field"
              name="role"
              type="radio"
              value="SELLER"
            />
            sign.seller
          </label>
        </div>
        <button
          class="sign__button"
          style="background-color: rgb(0, 0, 0);"
          type="submit"
        >
          sign.signUp
        </button>
      </form>
      <div
        class="sign__text-link"
      >
        sign.have
         
        <a
          class="text-link"
          href="/sign-in"
          style="color: rgb(0, 0, 0);"
        >
          sign.signIn
        </a>
      </div>
    </div>
  </div>
</div>
`;
